src/test/java/org/jenkinsci/plugins/workflow/support/steps/build/BuildTriggerStepTest.java

```diff
@@ -595,4 +598,49 @@ public void invalidChoiceParameterValue() throws Exception {
     public void buildStepDocs() throws Exception {
         SnippetizerTester.assertDocGeneration(BuildTriggerStep.class);
     }
+
+    @Test public void automaticParameterConversion() throws Exception {
+        WorkflowJob ds = j.createProject(WorkflowJob.class);
+        ds.addProperty(new ParametersDefinitionProperty(
+                new PasswordParameterDefinition("password", "default", "description"),
+                new BooleanParameterDefinition("boolean", false, "description")
+        ));
+        ds.setDefinition(new CpsFlowDefinition(
+                "echo('Password: ' + params.password)\n" +
+                "echo('Boolean: ' + params.boolean)\n", true));
+        WorkflowJob us = j.createProject(WorkflowJob.class);
+        String def = "build(job: '" + ds.getName() + "', parameters: [%s])";
+
+        // A password parameter passed as a string parameter is converted.
+        us.setDefinition(new CpsFlowDefinition(String.format(def, "string(name: 'password', value: 'secret')"), true));
+        WorkflowRun b1 = j.buildAndAssertSuccess(us);
+        j.assertLogContains("The parameter 'password' did not have the type expected", b1);
+        assertThat(getParameter(ds.getBuildByNumber(1), "password").getDescription(), equalTo(Messages.BuildTriggerStepExecution_convertedParameterDescription("description", "Password Parameter", us.getName() + " #1")));
+
+        // A password parameter passed as a text parameter is converted.
+        us.setDefinition(new CpsFlowDefinition(String.format(def, "text(name: 'password', value: 'secret')"), true));
+        WorkflowRun b2 = j.buildAndAssertSuccess(us);
+        j.assertLogContains("The parameter 'password' did not have the type expected", b2);
+        assertThat(getParameter(ds.getBuildByNumber(2), "password").getDescription(), equalTo(Messages.BuildTriggerStepExecution_convertedParameterDescription("description", "Password Parameter", us.getName() + " #2")));
+
+        // A password parameter passed as a password parameter is not converted.
+        us.setDefinition(new CpsFlowDefinition(String.format(def, "password(name: 'password', value: 'secret')"), true));
+        WorkflowRun b3 = j.buildAndAssertSuccess(us);
+        j.assertLogNotContains("The parameter 'password' did not have the type expected", b3);
```

Comment:
TODO: I want to add a test where a password parameter is passed as a boolean parameter to show that the boolean parameter is not converted.

Reply:
Done in a049a76.